packages/presentation/lib/ui/auth_page/auth_widget.dart

```diff
@@ -95,60 +95,43 @@ class _FormWidget extends StatelessWidget {
       child: Padding(
         padding: const EdgeInsets.symmetric(horizontal: Dimens.size26),
         child: Column(
-          mainAxisAlignment: MainAxisAlignment.center,
-          crossAxisAlignment: CrossAxisAlignment.start,
           children: [
             _ErrorMessageWidget(
               errorMessage: errorMessage.orEmpty,
             ),
-            Text(
-              S.of(context).userName,
-              style: TextStyles.sfProMed12(
-                color: ColorsApplication.colorProfileTitle,
-              ),
-            ),
-            const SizedBox(height: Dimens.size8),
-            TextField(
-              controller: bloc.textLoginController,
-              cursorColor: Colors.white,
-              style: const TextStyle(color: ColorsApplication.colorTitle),
-              decoration: InputDecoration(
-                prefixIcon: SvgPicture.asset(
-                  ImagesUtils.profile,
-                  width: Dimens.size18,
-                  height: Dimens.size18,
-                  fit: BoxFit.none,
-                ),
-                filled: true,
-                fillColor: ColorsApplication.colorTextField,
-              ),
-            ),
-            const SizedBox(height: Dimens.size20),
-            Text(
-              S.of(context).password,
-              style: TextStyles.sfProMed12(
-                color: ColorsApplication.colorProfileTitle,
-              ),
-            ),
-            const SizedBox(height: Dimens.size8),
-            TextField(
-              controller: bloc.textPasswordController,
-              obscureText: true,
-              autocorrect: false,
-              enableSuggestions: false,
-              cursorColor: Colors.white,
-              style: const TextStyle(
-                color: ColorsApplication.colorTitle,
-              ),
-              decoration: InputDecoration(
-                prefixIcon: SvgPicture.asset(
-                  ImagesUtils.lock,
-                  width: Dimens.size18,
-                  height: Dimens.size18,
-                  fit: BoxFit.none,
-                ),
-                filled: true,
-                fillColor: ColorsApplication.colorTextField,
+            Form(
+              key: bloc.formKey,
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Text(
+                    S.of(context).userName,
+                    style: TextStyles.sfProMed12(
+                      color: ColorsApplication.colorProfileTitle,
+                    ),
+                  ),
+                  const SizedBox(height: Dimens.size8),
+                  _TextFormField(
+                    validator: bloc.validatorLogin(context),
```

Comment:
seems like u invoke validation function right here while creating a formField and pass a string, which used in validate: () => String. further in bloc, u invoke formKey.currentState?.validate(), then in textForm should be invoked methods
validate: () => validator, where validator is this string, which u pass in constructor.
seems like it should not work correctly.
u can pass into your widgets constructor 
Function validate = bloc.validate()
and use it in your validate field

Reply:
11/10